pkg/operation/botanist/secrets.go

```diff
@@ -600,6 +574,65 @@ func (b *Botanist) generateWantedSecrets(basicAuthAPIServer *secrets.BasicAuth,
 		},
 	})
 
+	if b.Shoot.KonnectivityTunnelEnabled {
+		// TODO: remove when all Gardener supported versions are >= 1.18
+		vpnSecretNamesToDelete := []string{"vpn-seed", "vpn-seed-tlsauth", "vpn-shoot"}
+		for _, secret := range vpnSecretNamesToDelete {
+			if err := b.K8sSeedClient.Client().Delete(context.TODO(), &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secret, Namespace: b.Shoot.SeedNamespace}}); client.IgnoreNotFound(err) != nil {
+				return nil, err
+			}
+		}
```

Comment:
Same comment as https://github.com/gardener/gardener/pull/2251#discussion_r420538804, should be part of of another function